fine_tuning/evaluate_finetuned.py

```
@@ -0,0 +1,218 @@
+            temperature=0.0,  # Deterministic
+            response_format={"type": "json_object"},
+        )
+
+        raw_content = response.choices[0].message.content
+        try:
+            parsed = json.loads(raw_content)
+        except json.JSONDecodeError as exc:
+            raise ValueError(f"Invalid JSON response: {raw_content}") from exc
+
+        if not isinstance(parsed, dict) or "category" not in parsed:
+            raise ValueError(f"Unexpected response format: {parsed}")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)